src/conn/conn_log.c

```diff
@@ -314,6 +314,8 @@ __wt_log_truncate_files(
 
 	WT_UNUSED(cfg);
 	conn = S2C(session);
+	if (!FLD_ISSET(conn->log_flags, WT_CONN_LOG_ENABLED))
+		return (0);
```

Comment:
Thanks @keithbostic, I've made the style change.

I suspect we're seeing this now because I added diagnostic yields into `__wt_writelock`: that should help shake out some additional races (e.g., on open or close in this case).